src/app/api/swap/quote/route.ts

```diff
@@ -0,0 +1,92 @@
+import { NextRequest, NextResponse } from 'next/server'
+import { Address, isAddress, getAddress } from 'viem'
+import { uniswapProvider } from '@/lib/swap/providers/uniswap'
+import { SWAP_TOKEN_ADDRESSES } from '@/lib/swap/constants'
+import { getTokenDecimalsBatch, scaleAmount, isValidAmount } from '@/lib/swap/utils'
+
+/**
+ * Cache quotes for 30 seconds
+ * Quotes are time-sensitive, so we use a short cache duration
+ */
+export const revalidate = 30
+
+/**
+ * GET /api/swap/quote
+ *
+ * Query parameters:
+ * - tokenIn: Token address to swap from (defaults to USDT0)
+ * - tokenOut: Token address to swap to (defaults to USDRIF)
+ * - amount: Amount to swap (human-readable string, e.g., "100.5")
+ *
+ * Example:
+ * GET /api/swap/quote?tokenIn=0x779dED0C9e1022225F8e0630b35A9B54Be713736&tokenOut=0x3A15461d8AE0f0Fb5fA2629e9dA7D66A794a6E37&amount=100
+ */
+export async function GET(request: NextRequest) {
```

Comment:
if this is gonna be used only inside our app and not consumed outside , this could be converted to a Server action with "use server" directive. It's safer and more "modern" way in the Next app. Think about it next time. However for now this endpoint will also do its job fine.